app/src/main/java/com/x8bit/bitwarden/data/platform/datasource/di/PreferenceModule.kt

```diff
@@ -35,14 +37,55 @@ object PreferenceModule {
     fun provideEncryptedSharedPreferences(
         application: Application,
     ): SharedPreferences =
-        EncryptedSharedPreferences
-            .create(
-                application,
-                "${application.packageName}_encrypted_preferences",
-                MasterKey.Builder(application)
-                    .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
-                    .build(),
-                EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
-                EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM,
-            )
+        @Suppress("TooGenericExceptionCaught")
+        try {
+            getEncryptedSharedPreferences(application = application)
+        } catch (e: GeneralSecurityException) {
+            // Handle when a bad master key or key-set has been attempted
+            destroySharedPreferencesAndRebuild(application = application)
+        } catch (e: RuntimeException) {
+            // Handle KeystoreExceptions that get wrapped up in a RuntimeException
+            destroySharedPreferencesAndRebuild(application = application)
+        }
+
+    /**
+     * Completely destroys the keystore master key and encrypted shared preferences file. This will
+     * cause all users to be logged out since the access and refresh tokens will be removed.
```

Comment:
Have we confirmed what the behavior is if this actually happens when we have logged in users? I'm guessing this looks like a "soft logout", like what happens after you unlocking has failed multiple times?

Reply:
@brian-livefront It did look like a soft-logout but had additional decryption issues. I had to delete the regular shared preferences, in order to get the desired affect.

Do you think we are OK with this?